pygenn/model_preprocessor.py

```diff
@@ -312,7 +289,8 @@ def __init__(self, variable_name, variable_type, values=None):
         else:
             self.is_scalar = True
             self.type = variable_type
-
+        
+        self.group = proxy(group)
```

Comment:
what does proxy do?

Reply:
it's basically a weak reference to prevent circular-dependencies

Reply:
https://docs.python.org/3/library/weakref.html

Reply:
aaah, I remember you fixing other stuff with this :)